apps/astarte_appengine_api/lib/astarte_appengine_api/rpc/vmq_plugin.ex

```diff
@@ -22,38 +22,22 @@ defmodule Astarte.AppEngine.API.RPC.VMQPlugin do
   """
   require Logger
 
-  alias Astarte.RPC.Protocol.VMQ.Plugin, as: Protocol
-
-  alias Astarte.RPC.Protocol.VMQ.Plugin.{
-    Call,
-    GenericErrorReply,
-    GenericOkReply,
-    Publish,
-    PublishReply,
-    Reply
-  }
-
-  alias Astarte.AppEngine.API.Config
-
-  require Logger
-
-  @rpc_client Config.rpc_client!()
-  @destination Protocol.amqp_queue()
+  @rpc_behaviour Application.compile_env(
+                   :astarte_appengine_api,
+                   :vernemq_plugin_rpc_client,
+                   Astarte.AppEngine.API.RPC.VMQPlugin.Client
+                 )
 
   def publish(topic, payload, qos)
```

Comment:
The previous implementation of `publish/3` returned `:ok`, while now it returns:
- `:ok` in tests, since the `@rpc_behaviour` is mocked to return `:ok`.
- `{:ok, %{...}}` in production, when using the actual implementation for `@rpc_behavior`.

I wonder if callers of `publish/3` expect a `:ok` return value, and so tests pass but the code may break in production.
Let's double check this. And even if it's not the case, I would strive to have a single specific return value: `:ok` is fine, since it already was fine before.
Arguably, we could even have VerneMQ just return `:ok` if what is currently returning (`%{local_matches: ..., remote_matches: ...}`) is not of interest, but that's a separate issue.

Reply:
I would leave `{:ok, %{local_matches: integer(), remote_matches: integer()}}` as it was before, because the number of local/remote matches might become relevant in case of a netsplit